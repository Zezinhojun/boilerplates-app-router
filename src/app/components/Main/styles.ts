"use client";

import { Box, Typography } from "@mui/material";
import { styled } from "@mui/material/styles";

export const Wrapper = styled(Box)({
  backgroundColor: "#06092b",
  color: "#fff",
  width: "100%",
  height: "100%",
  padding: "3rem",
  textAlign: "center",
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  justifyContent: "center",
});

export const Logo = styled("img")({
  width: "25rem",
  marginBottom: "2rem",
});

export const Title = styled(Typography)({
  fontSize: "2.5rem",
});

export const Description = styled(Typography)({
  fontSize: "2rem",
  fontWeight: 400,
});

export const Illustration = styled("img")({
  marginTop: "3rem",
  width: "min(30rem, 100%)",
});
