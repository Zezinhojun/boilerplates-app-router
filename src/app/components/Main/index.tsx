"use client";

import * as S from "./styles";

const Main = ({
  title = "React AvanÃ§ado",
  description = "Typescript, ReactJS, NextJS e Styled Components",
}) => (
  <S.Wrapper>
    <S.Logo src="/img/logo.svg" />
    <S.Title variant="h1">{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration src="/img/hero-illustration.svg" />
  </S.Wrapper>
);

export default Main;
