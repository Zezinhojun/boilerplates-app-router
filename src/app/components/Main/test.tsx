import { screen, render } from "@testing-library/react";
import Main from ".";

describe("<Main />", () => {
  it("Should render the heading", () => {
    //renderiza o componente
    const { container } = render(<Main />);

    //busca o elemento e verifica a existência dele
    expect(
      screen.getByRole("heading", { name: /react avançado/i }),
    ).toBeInTheDocument();

    //gerar snapshot
    expect(container.firstChild).toMatchSnapshot();
  });
});
