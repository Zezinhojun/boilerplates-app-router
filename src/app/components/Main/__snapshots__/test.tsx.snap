// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<Main /> Should render the heading 1`] = `
<div
  class="MuiBox-root css-c2qu53"
>
  <img
    class="css-p803vk"
    src="/img/logo.svg"
  />
  <h1
    class="MuiTypography-root MuiTypography-h1 css-82eldw-MuiTypography-root"
  >
    React Avan√ßado
  </h1>
  <p
    class="MuiTypography-root MuiTypography-body1 css-slbfx3-MuiTypography-root"
  >
    Typescript, ReactJS, NextJS e Styled Components
  </p>
  <img
    class="css-5h5p0m"
    src="/img/hero-illustration.svg"
  />
</div>
`;
